app/adapters/application.js

```diff
@@ -18,7 +18,7 @@ async function retryOnError(ajax, ajaxArgs, retryCount = 0) {
       await sleep(250 * (retryCount + 1));
       return retryOnError(ajax, ajaxArgs, retryCount + 1);
     } else {
-      throw new Error(error);
+      throw error;
```

Comment:
EmberData throws specific Errors which we can use to determine what went wrong. Instead of wrapping it in a generic error instance, we return the specific one directly.